src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/Integrations/ConsumerConsumeSyncIntegration.cs

```diff
@@ -0,0 +1,96 @@
+// <copyright file="ConsumerConsumeSyncIntegration.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+using OpenTelemetry.AutoInstrumentation.CallTarget;
+using OpenTelemetry.AutoInstrumentation.DuckTyping;
+using OpenTelemetry.AutoInstrumentation.Instrumentations.Kafka.DuckTypes;
+using OpenTelemetry.Context.Propagation;
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.Kafka.Integrations;
+
+/// <summary>
+/// Kafka consumer sync consume instrumentation
+/// </summary>
+[InstrumentMethod(
+    assemblyName: IntegrationConstants.ConfluentKafkaAssemblyName,
+    typeName: IntegrationConstants.ConsumerTypeName,
+    methodName: IntegrationConstants.ConsumeSyncMethodName,
+    returnTypeName: IntegrationConstants.ConsumeResultTypeName,
+    parameterTypeNames: new[] { ClrNames.Int32 },
+    minimumVersion: IntegrationConstants.MinVersion,
+    maximumVersion: IntegrationConstants.MaxVersion,
+    integrationName: IntegrationConstants.IntegrationName,
+    type: InstrumentationType.Trace)]
+public static class ConsumerConsumeSyncIntegration
+{
+    internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, int timeout)
+    {
+        // Attempting to consume another message marks the end
+        // of a previous message processing
+        KafkaCommon.StopCurrentConsumerActivity();
+
+        return CallTargetState.GetDefault();
+    }
+
+    internal static CallTargetReturn<TResponse> OnMethodEnd<TTarget, TResponse>(TTarget instance, TResponse response, Exception? exception, in CallTargetState state)
```

Comment:
Since TTarget is always Confluent.Kafka.Consumer'2, you can always duck type (no chance of failure unless their API changes) and avoid doing it later. The only thing that changes is that now you'll need to change your call to the ConsumerCache so it's like: `ConsumerCache.TryGet(instance.Instance!, out var groupId)`

```suggestion
    internal static CallTargetReturn<TResponse> OnMethodEnd<TTarget, TResponse>(TTarget instance, TResponse response, Exception? exception, in CallTargetState state)
        where TTarget : IClientName, IDuckType
```

Reply:
It seems there is additional investigation needed as this approach doesn't work for me. I created an #3136 for tracking purposes.